### ISSUE TITLE
Repeated question about following links when opening status

### ISSUE THREAD

**Issue by magthe** [gekoke/magit-file-icons#2]
## Observed behaviour

When opening the magit  status, `magit-status`, the first time in a workspace I get the question

> Symbolic link to Git-controlled source file; follow link? (y or n)

I have to press `y` several times before I'm shown the magit status.

This happens for every repository I've tried so far, and it's repeated after restarting Emacs

## Desired behaviour

Not having the question come up at all :grin: 

## My config

My setup of `magit-file-icons` looks like this

```
(use-package magit-file-icons
  :after magit
  :init (magit-file-icons-mode 1))
```

**Comment by gekoke** [comment]
Hi, thanks for you report.

I'm having trouble reproducing. Can you specify what files you have in these repositories, or does that seem to not play a role? Are there actually any tracked symlinks?

I'm assuming that this doesn't happen when `magit-file-icons` isn't installed? How about when the minor mode is disabled?

The function that gives you that prompt is `vc-refresh-state`. In Magit, it's only called in [this line](https://github.com/magit/magit/blob/8b2d4b03ecf9635c165d1c0f90cd6f2eb415cafa/lisp/magit-files.el#L452). Actually, that seems to be the only call to any function from the `vc` package.

**Comment by magthe** [comment]
> I'm having trouble reproducing. Can you specify what files you have in these repositories, or does that seem to not play a role? Are there actually any tracked symlinks?

No, there are no tracked symlinks in the repositories where I see this behaviour.

> I'm assuming that this doesn't happen when `magit-file-icons` isn't installed? How about when the minor mode is disabled?

Indeed, it does not happen when `magit-file-icons` isn't installed. If the minor mode is disabled I don't get the question, but as soon as I enable the mode the question pops up.

**Comment by justinbarclay** [comment]
@magthe Are you using Elpaca? Or maybe straight? I also ran into this issue, and I think it's a combination of Elpaca symlinking packages `el-patch` not playing nicely. Unfortunately, short of the author removing `el-patch,` I am not sure much can be done about that message because `el-patch` seems to have to open up the base file _every_ time it is called.

**Comment by magthe** [comment]
I'm using straight.

**Comment by jumper047** [comment]
Also encountered this issue after moving to elpaca. This workaround works for me:
```
    (defun j47/execute-with-follow-symlinks (orig-fun &rest args)
      (let ((vc-follow-symlinks 't))
        (apply orig-fun args)))
    (advice-add 'el-patch-eval-template :around  #'j47/execute-with-follow-symlinks)
```
